tests/test_client.py

```diff
@@ -578,3 +595,357 @@ def test_history_from_json():
+                KSWorkflowParams(
+                    actions=[KSAction(result=KSResultAction("some result"))]
+                ),
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+            ),
+            spec=ScheduleSpec(),
+            state=ScheduleState(paused=True),
+        ),
+        trigger_immediately=True,
+    )
+
+    # Confirm workflow result
+    desc = await handle.describe()
+    assert desc.info.num_actions == 1
+    action_exec = (await handle.describe()).info.recent_actions[0].action
+    assert isinstance(action_exec, ScheduleActionExecutionStartWorkflow)
+    assert (
+        "some result"
+        == await client.get_workflow_handle(
+            action_exec.workflow_id, run_id=action_exec.first_execution_run_id
+        ).result()
+    )
+
+    await handle.delete()
+    await assert_no_schedules(client)
+
+
+async def test_schedule_backfill(
+    client: Client, worker: ExternalWorker, env: WorkflowEnvironment
+):
+    if env.supports_time_skipping:
+        pytest.skip("Java test server doesn't support schedules")
+    await assert_no_schedules(client)
+
+    # Just in case it's on the minute boundary, move it off
+    now = datetime.utcnow()
+    if now.second == 0:
+        now += timedelta(seconds=1)
+
+    # Create paused schedule that runs every minute and has two backfills
+    handle = await client.create_schedule(
+        f"schedule-{uuid.uuid4()}",
+        Schedule(
+            action=ScheduleActionStartWorkflow(
+                "kitchen_sink",
+                KSWorkflowParams(
+                    actions=[KSAction(result=KSResultAction("some result"))]
+                ),
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+            ),
+            spec=ScheduleSpec(
+                intervals=[ScheduleIntervalSpec(every=timedelta(minutes=1))],
+            ),
+            state=ScheduleState(paused=True),
+        ),
+        backfill=[
+            ScheduleBackfill(
+                start_at=now - timedelta(minutes=30),
+                end_at=now - timedelta(minutes=29),
+                overlap=ScheduleOverlapPolicy.ALLOW_ALL,
+            )
+        ],
+    )
+    assert 2 == (await handle.describe()).info.num_actions
+
+    # Add two more backfills and and -2m will be deduped
+    await handle.backfill(
+        ScheduleBackfill(
+            start_at=now - timedelta(minutes=4),
+            end_at=now - timedelta(minutes=2),
+            overlap=ScheduleOverlapPolicy.ALLOW_ALL,
+        ),
+        ScheduleBackfill(
+            start_at=now - timedelta(minutes=2),
+            end_at=now,
+            overlap=ScheduleOverlapPolicy.ALLOW_ALL,
+        ),
+    )
+    assert 6 == (await handle.describe()).info.num_actions
+
+    await handle.delete()
+    await assert_no_schedules(client)
+
+
+async def assert_no_schedules(client: Client) -> None:
+    # Listing appears eventually consistent
```

Comment:
nit: remove comment now that this is confirmed?

Reply:
That makes the comment even more important to explain why I'm repeatedly getting count